src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.ts

```diff
@@ -21,62 +21,92 @@ export const useRespondentAnswers = () => {
     startTime: providedStartTime,
     endTime: providedEndTime,
     filterByIdentifier: providedFilterByIdentifier,
+    identifier: providedIdentifier,
     versions: providedVersions,
+    isIdentifiersChange = false,
   }: FetchAnswers) => {
     if (!appletId || !respondentId) return;
+
     try {
       const {
-        startDate,
+        startDate: formStartDate,
         endDate: formEndDate,
         startTime: formStartTime,
         endTime: formEndTime,
-        identifier,
+        identifier: formIdentifier,
         filterByIdentifier: formFilterByIdentifier,
         versions: formVersions,
         identifiers,
       } = getValues();
-      const endDate = providedEndDate ?? formEndDate;
       const startTime = providedStartTime ?? formStartTime;
       const endTime = providedEndTime ?? formEndTime;
       const filterByIdentifier = providedFilterByIdentifier ?? formFilterByIdentifier;
       const versions = providedVersions ?? formVersions;
+      const identifier = providedIdentifier ?? formIdentifier;
+
+      const { selectedIdentifiers, recentAnswerDateString } =
+        getIdentifiers({
+          filterByIdentifier,
+          filterIdentifiers: identifier,
+          identifiers,
+        }) ?? {};
+
+      const {
+        identifierAnswerStartDate,
+        identifierAnswerEndDate,
+        activityAnswerStartDate,
+        activityAnswerEndDate,
+      } =
+        processIdentifiersChange({
+          setValue,
+          isIdentifiersChange,
+          adjustStartEndDates: !!(filterByIdentifier && identifier?.length),
+          recentIdentifiersAnswerDate: recentAnswerDateString,
+          activityLastAnswerDate: activity.lastAnswerDate,
+        }) ?? {};
 
-      const selectedIdentifiers = getIdentifiers(filterByIdentifier, identifier, identifiers);
+      const startDate = identifierAnswerStartDate ?? activityAnswerStartDate ?? formStartDate;
```

Comment:
Could you please make sure that tests cover the cases:
- when `identifierAnswerStartDate` is undefined/null?
- when `identifierAnswerStartDate` and `activityAnswerStartDate` are undefined/null?

Please check the same for endDate's fallbacks.

Reply:
It's already teste in the first test case:
![image](https://github.com/ChildMindInstitute/mindlogger-admin/assets/116339462/62f034f0-bdb7-4162-9ab0-7b7c4300ce3e)